### PR TITLE
Update Azure Linux 3 base image source

### PR DESCRIPTION
**Pull request by mboersma**
## Change description

Updates the `image_offer` slightly to match the source of Azure Linux now that version 3 is released. Building on the preview base began to fail recently due to missing rpms, which makes sense.

Also defaults to testing Azure Linux 3 instead of version 2 (Mariner) ~~and adds a CVM target~~.

## Related issues

- Fixes #

## Additional context

You can poke around at what images are available with this Azure CLI command:
```shell
 az vm image list --publisher MicrosoftCBLMariner --all
```

### CODE REVIEW

**Comment by mboersma** [comment]
`azurelinux-3-cvm` failed with this error:

```
[0;32m    azure-arm.sig-azurelinux-3-cvm: TASK [kubernetes : Install Kubernetes] *****************************************[0m
[0;32m    azure-arm.sig-azurelinux-3-cvm: fatal: [default]: FAILED! => {"changed": false, "failures": [], "msg": "Unknown Error occurred: Transaction test error:\n  installing package kubeadm-1.30.5-150500.1.1.x86_64 needs 6MB more space on the / filesystem\n  installing package kubelet-1.30.5-150500.1.1.x86_64 needs 106MB more space on the / filesystem\n  installing package kubectl-1.30.5-150500.1.1.x86_64 needs 158MB more space on the / filesystem\n\nError Summary\n-------------\nDisk Requirements:\n   At least 158MB more space needed on the / filesystem.\n", "rc": 1, "results": []}[0m
```

**Comment by mboersma** [comment]
/retitle [WIP] Update Azure Linux 3 base image source

**Comment by mboersma** [comment]
/retitle Update Azure Linux 3 base image source

File: images/capi/ansible/roles/setup/tasks/azurelinux.yml [hunk 1]
```diff
@@ -14,20 +14,20 @@
 ---
 - ansible.builtin.import_tasks: rpm_repos.yml
 
-- name: Perform a yum update
-  ansible.builtin.yum:
+- name: Perform a tdnf update
```
Comment (drew-viles) [inline images/capi/ansible/roles/setup/tasks/azurelinux.yml:17]: tdnf? or is it a typo?

Comment (drew-viles) [reply images/capi/ansible/roles/setup/tasks/azurelinux.yml:17]: Nope, just wanted to check it wasn't a typo :-)

**Pull request merged**